src/api/clock.rs

```diff
@@ -0,0 +1,52 @@
+use embedded_time::TimeInt;
+use embedded_time::fraction::Fraction;
+
+#[derive(Clone, Copy, PartialOrd, Ord, Eq, fmt::Debug)]
```

Comment:
We shouldn't derive `PartialOrd` and `Ord`. If we need those traits, we should implement it ourselves because the derived implementation is wrong.

Reply:
I do not understand why the derived implementation is wrong. Using our creation functions (` new()`, `new_wrap()`) we are assured that only the 24 LSB of the u32 are non-zero. Therefore, it looks safe to me that `PartialOrd` and `Ord`  hold. Do I miss anything?